### PR TITLE
Doc with plans for converting core to C++

### PR DESCRIPTION
**Pull request by vjpai**
From a discussion with @ctiller and @markdroth

### CODE REVIEW

File: doc/core/moving-to-c++.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Moving gRPC core to C++
+
+October 2017
+
+ctiller, markdroth, vjpai
+
+## Background and Goal
+
+gRPC core was originally written in C89 for several reasons (possibility of
```
Comment (markdroth) [inline doc/core/moving-to-c++.md]: Might be useful to add a link to https://github.com/grpc/proposal/blob/master/L6-allow-c%2B%2B-in-grpc-core.md for context.  Although it looks like the plan we discussed on Friday is somewhat different from the one described in the gRFC.

Comment (vjpai) [reply doc/core/moving-to-c++.md]: Sounds good, I did that.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by vjpai** [comment]
Merging early to get this publicized, since docs only and sanity passes.

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__2M_1_0.opt.new: 1


[microbenchmarks] No significant performance differences
```

**Pull request merged**